src/loctool/PendoXliffFileType.ts

```diff
@@ -0,0 +1,347 @@
+        /* eslint-disable-next-line
+         @typescript-eslint/no-unsafe-assignment,
+         @typescript-eslint/no-unsafe-call,
+         @typescript-eslint/no-explicit-any,
+         @typescript-eslint/no-unsafe-member-access -- yet undocumented utility method,
+         signature based on https://github.com/iLib-js/ilib-loctool-tap-i18n/blob/7585e97497e16475bfce1fc034caf0c7716229e1/YamlFileType.js#L122 */
+        const fileLocale: string = (this.loctoolAPI.utils as any).getLocaleFromPath(
+            pathMapping.template,
+            pathInProject,
+        );
+        if (fileLocale !== this.sourceLocale) {
+            return false;
+        }
+
+        return true;
+    }
+
+    write(): void {
+        // no-op
+        // per loctool convention, when localized files are written individually
+        // there is no need to implement this method
+        // (which is meant to write out aggregated resources - whatever that means);
+        // this plugin intends to take a source XLIFF file on the input and from it
+        // parse resources which have the source string transformed (markdown syntax escaped)
+        // while on the output, it should produce one copy of the original XLIFF file
+        // for each target locale (with the translated string backconverted from the escaped syntax
+        // using the original non-escaped source string as a reference)
+    }
+
+    /**
+     * Collection of instantiated {@link PendoXliffFile}s by this file type.
+     *
+     * See {@link getExtracted} on why this is needed.
+     */
+    private readonly files: Record<string, PendoXliffFile> = {};
+
+    newFile(pathInProject: string): PendoXliffFile {
+        if (this.files[pathInProject]) {
+            throw new Error(`Attempt to create a file that already exists: ${pathInProject}`);
+        }
+
+        const actualPath = path.join(this.project.getRoot(), pathInProject);
+
+        // inject localized path generator, translation set creation and locale mapping
+        // into the file instance capturing parameters from the current context where needed
+        //
+        // (this is done to decouple file processing logic from plugin-related logic
+        // (accessing loctool API, project, configuration etc.)
+
+        // register the file instance
+        this.files[pathInProject] = new PendoXliffFile(
+            actualPath,
+            (loctoolLocale) => path.join(this.project.getRoot(), this.getLocalizedPath(pathInProject, loctoolLocale)),
+            (loctoolLocale) => this.getOuputLocale(loctoolLocale),
+            (units) => this.createTranslationSet(units),
+        );
+
+        return this.files[pathInProject];
+    }
+
+    /**
+     * Default path mappings for Pendo XLIFF files.
+     */
+    private static readonly defaultPathMappings: Record<string, { template: string }> = {
+        "**/*.xliff": {
+            template: "[dir]/[basename]_[locale].xl(if)?f",
```

Comment:
The template is not a regular expression, so `(if)?` will not work here.
```suggestion
            template: "[dir]/[basename]_[locale].xliff",
```

Reply:
Fixed in 8cecd70